app/org/sagebionetworks/bridge/models/schedules/ScheduleCriteria.java

```diff
@@ -1,92 +1,37 @@
 package org.sagebionetworks.bridge.models.schedules;
 
+import static com.google.common.base.Preconditions.checkNotNull;
+
 import java.util.Objects;
-import java.util.Set;
 
 import org.sagebionetworks.bridge.models.Criteria;
 
 import com.fasterxml.jackson.annotation.JsonCreator;
 import com.fasterxml.jackson.annotation.JsonProperty;
-import com.google.common.collect.Sets;
 
-public final class ScheduleCriteria implements Criteria {
+public final class ScheduleCriteria {
     private final Schedule schedule;
-    private final Integer minAppVersion;
-    private final Integer maxAppVersion;
-    private final Set<String> allOfGroups;
-    private final Set<String> noneOfGroups;
+    private final Criteria criteria;
     
     @JsonCreator
-    private ScheduleCriteria(@JsonProperty("schedule") Schedule schedule, 
-            @JsonProperty("minAppVersion") Integer minAppVersion, 
-            @JsonProperty("maxAppVersion") Integer maxAppVersion, 
-            @JsonProperty("allOfGroups") Set<String> allOfGroups, 
-            @JsonProperty("noneOfGroups") Set<String> noneOfGroups) {
+    public ScheduleCriteria(@JsonProperty("schedule") Schedule schedule, 
+            @JsonProperty("criteria") Criteria criteria) {
+        checkNotNull(schedule);
+        checkNotNull(criteria);
+        
```

Comment:
This is user-submitted. If the user submits a ScheduleCriteria without a schedule or a criteria, this will throw a 500 Internal Server Error. It would be better to move this logic to the validator.